frontend/src/concepts/pipelines/content/tables/pipelineRun/PipelineRunTable.tsx

```diff
@@ -253,8 +322,56 @@ const PipelineRunTable: React.FC<PipelineRunTableProps> = ({
           }}
         />
       )}
+      {isCustomColModalOpen && (
+        <CustomMetricsColumnsModal
+          key={metricsNames.size}
+          experimentId={experimentId}
+          columns={[...new Set([...metricsColumnNames, ...metricsNames])].map((metricName) => ({
+            id: metricName,
+            content: metricName,
+            props: { checked: metricsColumnNames.includes(metricName) },
+          }))}
+          onClose={() => setIsCustomColModalOpen(false)}
+        />
+      )}
     </>
   );
 };
 
+const PipelineRunTable: React.FC<PipelineRunTableProps> = ({ runs, page, ...props }) => {
+  const [runArtifacts, runArtifactsLoaded, runArtifactsError] = useGetArtifactsByRuns(runs);
+  const metricsNames = new Set<string>();
+
+  const runsWithMetrics = runs.map((run) => ({
+    ...run,
+    metrics: runArtifacts.reduce((acc: ArtifactProperty[], runArtifactseMap) => {
+      const artifacts = Object.entries(runArtifactseMap).find(
+        ([runId]) => run.run_id === runId,
+      )?.[1];
+
+      artifacts?.forEach((artifact) => {
+        if (artifact.getType() === ArtifactType.METRICS) {
+          const artifactProperties = getArtifactProperties(artifact);
+
+          artifactProperties.map((artifactProp) => metricsNames.add(artifactProp.name));
+          acc.push(...artifactProperties);
+        }
+      });
+
+      return acc;
+    }, []),
+  }));
+
+  return (
+    <PipelineRunTableInternal
+      runs={runsWithMetrics}
+      page={page}
+      metricsNames={metricsNames}
+      artifactsLoaded={runArtifactsLoaded && !!runArtifacts.length}
```

Comment:
!!runArtifacts.length this is conflicting. should just be the loaded var

Reply:
We can't rely on loaded since it is always true after initially loading. Checking the length of runs fixes[ the issue](https://github.com/opendatahub-io/odh-dashboard/pull/2888#discussion_r1631620603).

Reply:
there is a loading issue tho where the empty pop up never shows becuase we say it is not loaded yet

Reply:
Empty pop up? I don't know what you mean.

Reply:
on hover of the icon button to open the modal. it shows a tooltip. and when it is empty it just remains on the loading text

Reply:
I'm not experiencing the same. Can you demonstrate this or are you assuming based on reading the code?

![image](https://github.com/opendatahub-io/odh-dashboard/assets/96431149/571c41c2-6f4a-4436-b0ec-438497f870f6)

Reply:
i will double check

Reply:
ope you are right